### ISSUE TITLE
[Due for payment 2025-04-03] [$250] Update NewDot report action copy to make it clear when the action is automatically (vs manually) initiated

### ISSUE THREAD

**Issue by maddylewis** [Expensify/App#53386]
## We currently don't show when a report was auto-exported in NewDot
Similarly to OldDot, we should differentiate when a report action is done automatically by Expensify or manually by the user account  

## Example Below 
EXFY Accounts Payable automatically exported this report to NetSuite, but the copy doesn't specify:
<img width="1138" alt="image" src="https://github.com/user-attachments/assets/e3c0ac08-cb98-42f4-8595-215833045a76">
<img width="1257" alt="image" src="https://github.com/user-attachments/assets/e2fb4cfb-9eae-44d2-b72b-630fc906357b">

## Update Copy
Let's update the copy to more clearly identify when something is manually vs. automatically exported. We should leave the copy for manual exporting unchanged. But if it is automatically exported, we should use:

>%PreferredExporterName% %time exported%
automatically exported this report to QuickBooks Online via [accounting settings](link to specific help page for the applicable integration explaining auto-sync)

We should also leave the link(s) to the specific expenses that currently exist at the end of the system message unchanged when automatically exporting. 

Additional input on copy changes - https://expensify.slack.com/archives/C07NZ8B1VTQ/p1733152728274999?thread_ts=1731949285.192699&cid=C07NZ8B1VTQ
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021864461454721236879</li>
        <li>Upwork Job ID: 1864461454721236879</li>
        <li>Last Price Increase: 2025-01-16</li>
<li>Automatic offers: </li>
<ul>
<li>twilight2294 | Contributor | 105912429</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @joekaufmanexpensify</details>

**Comment by joekaufmanexpensify** [comment]
Making sure we [are aligned](https://expensify.slack.com/archives/C07NZ8B1VTQ/p1733180551424719?thread_ts=1731949285.192699&cid=C07NZ8B1VTQ) on copy before opening this up for proposals.

**Comment by joekaufmanexpensify** [comment]
Updated based on slack discussion!

**Comment by joekaufmanexpensify** [comment]
Can you not see them at all? They aren't super high-res but i can still see all the words fine on my end.

**Comment by maddylewis** [comment]
@joekaufmanexpensify lmk if you need new screenshots. we could probs just ask jenna (those are her screenshots in the OP) to take a few more for us to add to the issue.

**Comment by joekaufmanexpensify** [comment]
>@joekaufmanexpensify, the images in the OP isn't uploaded correctly I guess, can you please update or share them?

@Krishna2323 I updated the images in OP!

**Comment by joekaufmanexpensify** [comment]
@alitoshmatov could you please review the proposals here?

**Comment by joekaufmanexpensify** [comment]
Bumped in Slack

**Comment by alitoshmatov** [comment]
@joekaufmanexpensify Just to clarify expected behavior did you mean like this(below image). Also
1. What does `[account settings]` means here?
2. Does (link to specific help page for the applicable integration explaining auto-sync) mean `View out-of-pocket expenses.` here? Or is it another additional link?

<img width="747" alt="Screenshot 2024-12-10 at 11 56 34 PM" src="https://github.com/user-attachments/assets/8f102013-14d1-4435-b365-ff8657af7c28">

**Comment by joekaufmanexpensify** [comment]
>What does [accounting settings] means here?
Good Q! `Accounting settings` is where someone will set whether automatic export is used. It is meant to link to help documentation that explains this. Specifically, one of these four links depending on the integration: 

- [QBO ](https://help.expensify.com/articles/new-expensify/connections/quickbooks-online/Configure-Quickbooks-Online#step-3-configure-advanced-settings)
- [NetSuite ](https://help.expensify.com/articles/new-expensify/connections/netsuite/Configure-Netsuite#step-3-configure-advanced-settings)
- [Xero ](https://help.expensify.com/articles/new-expensify/connections/xero/Configure-Xero#step-3-configure-advanced-settings)
- [Intacct](https://help.expensify.com/articles/new-expensify/connections/sage-intacct/Configure-Sage-Intacct#auto-sync)

>Does (link to specific help page for the applicable integration explaining auto-sync) mean View out-of-pocket expenses here? Or is it another additional link?
Another link! I shared the applicable links above. We don't want to change the existing links out to the expenses exported referenced here. This means if auto-export is used, there can be two links in the system message. 

LMK if that makes sense and if you have any other questions!

**Comment by alitoshmatov** [comment]
I think I got it. 

<img width="712" alt="Screenshot 2024-12-12 at 12 06 35 AM" src="https://github.com/user-attachments/assets/63325eb0-f391-4d04-b64e-9151801dca93" />

The final message should look like this and `account setting` here should redirect to one of this links based on which integration is used

- [QBO](https://help.expensify.com/articles/new-expensify/connections/quickbooks-online/Configure-Quickbooks-Online#step-3-configure-advanced-settings)
- [NetSuite](https://help.expensify.com/articles/new-expensify/connections/netsuite/Configure-Netsuite#step-3-configure-advanced-settings)
- [Xero](https://help.expensify.com/articles/new-expensify/connections/xero/Configure-Xero#step-3-configure-advanced-settings)
- [Intacct](https://help.expensify.com/articles/new-expensify/connections/sage-intacct/Configure-Sage-Intacct#auto-sync)

cc: @twilight2294 @NJ-2020 Can you update your proposal based on this

**Comment by joekaufmanexpensify** [comment]
@alitoshmatov could you share an update on the proposal review?

**Comment by alitoshmatov** [comment]
> URLofHelpDotSite will be taken based on the type of accounting method used, we can use the label itself to determine the type of connection and update the URLofHelpDotSite accordingly

@twilight2294 I am curious, how exactly are you going to handle this. I see that label is string, are you sure we will receive the same set of string always?

**Comment by alitoshmatov** [comment]
@twilight2294 I understand this, I mean do we have the list of all available lists and are we sure they always be the same. In example you provided there is `CONST.COMPANY_CARD.FEED_BANK_NAME`, do we have similar constant to rely on for our use case

**Comment by joekaufmanexpensify** [comment]
@alitoshmatov do you think we are close to selecting a proposal here?

**Comment by alitoshmatov** [comment]
No, I think based on how our code is setup we cannot show link to help page to corresponding integration. Because the are no distinct keys or values that we are receiving to for integration specifically, we only have integration name in message text.

If we are to apply solution based on current logic, we need an internal engineer to confirm if there is any way we could extract integration name out of current messages

**Comment by joekaufmanexpensify** [comment]
Interesting. So it is the piece of showing the help pages specifically that is making it difficult to proceed?

**Comment by alitoshmatov** [comment]
> Interesting. So it is the piece of showing the help pages specifically that is making it difficult to proceed?

@joekaufmanexpensify Yes

**Comment by alitoshmatov** [comment]
> It is not difficult @alitoshmatov , with my provided solution, we would show the correct links for correct cases

I understand what you mean here. But we are not 100% sure the label string will be having the same constants every time like here. We are not aware of the implementation of `label`. I am just asking for internal engineer who can confirm that we can reliably get integration type from this `label`

**Comment by joekaufmanexpensify** [comment]
>I am just asking for internal engineer who can confirm that we can reliably get integration type from this label
@alitoshmatov Is this something you still need help with?

**Comment by joekaufmanexpensify** [comment]
Bumping back to daily so we can find a volunteer for this.

**Comment by maddylewis** [comment]
@joekaufmanexpensify - do we need an internal engineer to review this one?

**Comment by alitoshmatov** [comment]
@joekaufmanexpensify Yes we need internal engineer just to confirm the following:
***

We are getting this message in automatic action
<img width="484" alt="Screenshot 2025-01-06 at 11 53 17 PM" src="https://github.com/user-attachments/assets/76bd64da-5023-4433-bebd-4471f5add52d" />

This comes from here 
https://github.com/Expensify/App/blob/787643fbddc8a4063e87f9a03da78dd7a23e991e/src/libs/ReportActionsUtils.ts#L1670

So integration name comes from `originalMessage.label` , do we have any constants to match this name against so that we can show corresponding help page. We have this help pages write now:
[QBO](https://help.expensify.com/articles/new-expensify/connections/quickbooks-online/Configure-Quickbooks-Online#step-3-configure-advanced-settings)
[NetSuite](https://help.expensify.com/articles/new-expensify/connections/netsuite/Configure-Netsuite#step-3-configure-advanced-settings)
[Xero](https://help.expensify.com/articles/new-expensify/connections/xero/Configure-Xero#step-3-configure-advanced-settings)
[Intacct](https://help.expensify.com/articles/new-expensify/connections/sage-intacct/Configure-Sage-Intacct#auto-sync)

We found this constant but I don't think it always matches label and we do not have help page for couple of this integrations.
https://github.com/Expensify/App/blob/126d92ab5cbb401efc0a5527faba244d04d48971/src/CONST.ts#L2595-L2604

**Comment by joekaufmanexpensify** [comment]
Still discussing in above thread.

**Comment by joekaufmanexpensify** [comment]
Still aligning [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1736547370581489?thread_ts=1736263952.629489&cid=C01GTK53T8Q).

**Comment by joekaufmanexpensify** [comment]
We're still figuring out the last outstanding question in Slack. I will make sure to update the issue as soon as it is resolved!

**Comment by joekaufmanexpensify** [comment]
Bumped [slack discussion](https://expensify.slack.com/archives/C03TQ48KC/p1737037482126399?thread_ts=1736863411.209879&cid=C03TQ48KC) to try and find internal engineer to help unblock this question.

**Comment by alitoshmatov** [comment]
@twilight2294 Can you update your proposal based on this https://expensify.slack.com/archives/C01GTK53T8Q/p1737038267708899?thread_ts=1736263952.629489&cid=C01GTK53T8Q

**Comment by alitoshmatov** [comment]
@twilight2294 I think we should create separate constant to store available labels. Also it looks like some integrations does not have their own doc so we should skip linking any url. One more thing the code should also consider possibility of label being unknown value.

I think we can go with @twilight2294 's [proposal](https://github.com/Expensify/App/issues/53386#issuecomment-2522134275)

C+ reviewed 🎀 👀 🎀

**Comment by cead22** [comment]
@alitoshmatov @twilight2294 I think that proposal needs to be updated:
- "I think we should create separate constant to store available labels": let's specify what that variable will be, its name, the values in it
- What integrations don't have their own doc? I think we should link all the ones we can
- Include how we'll handle the possibility of label being unknown value

The proposal says we'll update the code to the following

```
 exportedToIntegration: {
                    automaticOne: ({label} : ExportedToIntegrationParams) => `automatically exported this report to ${label} via`,
                    automaticTwo: 'accounting settings',
```

But 
- This copy is wrong `automatically exported this report to ${label} via`
- I think this copy is wrong `accounting settings`
- If `automaticOne` is for automatic and `automaticTwo` for manual, let's call them `automatic` and `manual`


The proposal says we'll update the code to the following
```
        result.push({
            text: Localize.translateLocal('report.actions.type.exportedToIntegration.automaticOne', {label}),
            url: '',
        });
        result.push({
            text: Localize.translateLocal('report.actions.type.exportedToIntegration.automaticTwo'),
            url: URLofHelpDotSite,
        })
```

But looking at the file, that looks wrong. Is it missing an `else if`?

> `URLofHelpDotSite` will be taken based on the type of accounting method used, we can use the `label` itself to determine the type of connection and update the `URLofHelpDotSite` accordingly

Let's add all the details related to this

**Comment by cead22** [comment]
> exportedToIntegration: {
>                     automaticOne: ({label} : ExportedToIntegrationParams) => `automatically exported this report to ${label} via`,
>                     automaticTwo: 'accounting settings',

> const url = linkURLofHelpDotSite[label];
>         result.push({
>             text: Localize.translateLocal('report.actions.type.exportedToIntegration.automaticTwo'),
>             url: url || '',
>         })

The proposal doesn't look right
- `automaticTwo` has no URL param, and the copy isn't producing a link
- Why do we need two separate copies `automaticOne`, and `automaticTwo`?

**Comment by cead22** [comment]
> where `automaticTwo` will contain the text `accounting settings` and it will be hyperlinked to the helpdot page by the URL variable of that respective integration.

Shouldn't `automaticTwo` be set to something like this?
```js
automaticOne: ({label} : ExportedToIntegrationParams) => automatically exported this report to ${label} via,
automaticTwo: ({url} : SomeType) => '[accounting settings](url)',
```

> Now we only want to hyperlink `accounting settings` text to the helpdot page, for that we would need to divide the text into two parts

Why do we need to divide the text into two parts instead of doing something like this
```js
automatic: ({label, url} : AutomaticExportToIntegrationParams) => `
	automatically exported this report to ${label} via ${url ? '[accounting settings](url)' : 'account settings'},
```

**Comment by cead22** [comment]
Oh I see, thanks for clarifying. @alitoshmatov does the updated proposal look good to you?

**Comment by alitoshmatov** [comment]
Yes I think the proposal is complete now and we can proceed.

**Comment by joekaufmanexpensify** [comment]
Awesome. @cead22 good with you to proceed as well?

**Comment by cead22** [comment]
Issue is assigned and no longer overdue

**Comment by joekaufmanexpensify** [comment]
TY!

**Comment by joekaufmanexpensify** [comment]
>@joekaufmanexpensify can you help me with the spanish translations here please, not able to login to slack.
>en: automatically exported this report to [Integration Name] via [accounting settings](URL)

@twilight2294 Yes! [Asked internally](https://expensify.slack.com/archives/C21FRDWCV/p1738691475773249).

**Comment by joekaufmanexpensify** [comment]
@twilight2294 it's

>EN: Automatically exported this report to [Integration Name] via accounting settings
>ES: Exportó automáticamente este informe a [Integration Name] a través de la configuración contable

**Comment by joekaufmanexpensify** [comment]
@twilight2294 do we think the PR will be up here soon?

**Comment by joekaufmanexpensify** [comment]
PR in review

**Comment by joekaufmanexpensify** [comment]
Same.

**Comment by alitoshmatov** [comment]
FYI I tested it using my real ND account

| Before  | After  |
|---------|--------|
| <img width="641" alt="Image" src="https://github.com/user-attachments/assets/2ccdbb54-1644-4496-9ad6-c7b2d62f36e1" /> |<img width="817" alt="Image" src="https://github.com/user-attachments/assets/5d9faf0a-1f1b-4a63-a347-67f1748729db" />|

**Comment by joekaufmanexpensify** [comment]
>Hey @joekaufmanexpensify @maddylewis what are the QA steps to test the PR? I tested the https://github.com/Expensify/App/pull/56470 using onyx mock data, but not sure how QA can test it, can you help me outline the steps ?

@twilight2294 Are the steps with mock data different? Or do you mean whether QA has access to actual accounting integration accounts to test with? If the latter, they do, so that is not a concern.

**Comment by joekaufmanexpensify** [comment]
Sounds good!

**Comment by joekaufmanexpensify** [comment]
PR pending internal review

**Comment by alitoshmatov** [comment]
@joekaufmanexpensify One small note about the result, auditor actions are also specified as automatic and are getting treated the same way. Is this expect?

<img width="938" alt="Image" src="https://github.com/user-attachments/assets/c8cf3234-0a7b-49d1-9954-98391ea3dd4c" />

**Comment by joekaufmanexpensify** [comment]
@alitoshmatov I am not 100% sure I follow the question. Are you asking why the highlighted box above says `automatically exported`? If so, could you share where that report comes from?

**Comment by alitoshmatov** [comment]
@joekaufmanexpensify This is from my workspace chat that I use to request payment for my C+ compensation(report id: `5050650694284275`). 

I might be onto nothing. I just wanted to make sure that expected behavior correct. 
| Before | After | 
|----------|----------|
| <img width="1138" alt="Image" src="https://github.com/user-attachments/assets/da18630a-1fed-44ae-9172-07dfd3884c9f" /> | <img width="1139" alt="Image" src="https://github.com/user-attachments/assets/97e286bb-1c3f-4bf2-ad73-4a1482448e8d" /> |

**Comment by joekaufmanexpensify** [comment]
Ah, I see. So you're saying that the changes in this PR will make it so exports not done with an accounting integration (like All data - report level export) will always also be listed as automatically performed? 

If so, that does seem odd to me. I think we'd want the export to be listed as automatically done only when that is the case.

**Comment by joekaufmanexpensify** [comment]
@alitoshmatov bump on the above. Curious if my understanding is correct?

**Comment by joekaufmanexpensify** [comment]
I see PR is also out on staging.

**Comment by alitoshmatov** [comment]
> Ah, I see. So you're saying that the changes in this PR will make it so exports not done with an accounting integration (like All data - report level export) will always also be listed as automatically performed?
> 
> If so, that does seem odd to me. I think we'd want the export to be listed as automatically done only when that is the case.

Sorry I missed your message. Yes that was my concern. So the PR is already merged. I should have handled this in better way. 

The problem was that it is not clear when it is done from integration, they both were being assigned automatic message based on this existing [logic](https://github.com/Expensify/App/blob/ece826f7eead51d03761ae36f039f64df6fe72f1/src/libs/ReportActionsUtils.ts#L1655).

**Comment by joekaufmanexpensify** [comment]
Got it. So going forward, is this what's going to happen:

- A manual export done via an accounting integration will not include the "automatic" part of the system message. 
- An automatic export done via an accounting integration will include the "automatic" part of the system message (we want this). 
- A manual export done via an export file, but not an accounting integration, will also include the "automatic" part of the system message (we don't want this). 

Is that a correct summary of what the PR does? If so, I will raise internally to see if we want to adjust the third point.

**Comment by joekaufmanexpensify** [comment]
@alitoshmatov Got it. So going forward, is this what's going to happen:

- A manual export done via an accounting integration will not include the "automatic" part of the system message.
- An automatic export done via an accounting integration will include the "automatic" part of the system message (we want this).
- A manual export done via an export file, but not an accounting integration, will also include the "automatic" part of the system message (we don't want this).

Is that a correct summary of what the PR does? If so, I will raise internally to see if we want to adjust the third point.

**Comment by alitoshmatov** [comment]
> Is that a correct summary of what the PR does? If so, I will raise internally to see if we want to adjust the third point.

I think so, but I'm not very knowledgeable about the difference between them.

**Comment by joekaufmanexpensify** [comment]
>I think so, but I'm not very knowledgeable about the difference between them.

Sounds good. I will test this to see if the manual export case works as I outlined above and report back.

**Comment by joekaufmanexpensify** [comment]
@alitoshmatov @twilight2294 I tested manually exporting to Quickbooks Online and the copy said the report was automatically exported. So I don't the solution here is working as expected. Can you guys take a look?

<img width="1511" alt="Image" src="https://github.com/user-attachments/assets/477ac552-761f-4516-a23b-261bf95a348c" />

**Comment by joekaufmanexpensify** [comment]
Yep confirmed this did not work for the manual case either. We are saying it was automatically exported, which is not correct. 

<img width="1511" alt="Image" src="https://github.com/user-attachments/assets/f543f13f-5e8e-4f7a-9696-3d5742fd2737" />

**Comment by joekaufmanexpensify** [comment]
[Discussing](https://expensify.slack.com/archives/C01GTK53T8Q/p1742485927627319)

**Comment by joekaufmanexpensify** [comment]
Removing payment hold, as we will pay this once the last PR is done.

**Comment by joekaufmanexpensify** [comment]
@alitoshmatov could you handle checklist so we can prep to pay this week?

**Comment by alitoshmatov** [comment]
**Regression Test Proposal**

1. Create expense
2. Pay it
3. Automatically export it with accounting integration
4. Make sure the action message is in the following format: 
 _automatically exported this report to QuickBooks Online via [accounting settings]_
5. [accounting settings] should be a link to a help dot page for corresponding accounting integration.

Do we agree 👍 or 👎

**Comment by joekaufmanexpensify** [comment]
Regression test issue here: https://github.com/Expensify/Expensify/issues/487647

**Comment by joekaufmanexpensify** [comment]
All set to issue payment. We need to pay:

- @twilight2294 $250 for PR via Upwork. 
- @alitoshmatov $250 for C+ via NewDot.

**Comment by joekaufmanexpensify** [comment]
@twilight2294 $250 sent and contract ended!

**Comment by joekaufmanexpensify** [comment]
@alitoshmatov Please request payment at your earliest convenience. Closing as this is otherwise all set!

**Comment by JmillsExpensify** [comment]
$250 approved for @alitoshmatov
